app/controllers/api/plan_copies_controller.rb

```diff
@@ -19,11 +19,18 @@ def create
     end
 
     respond_to do |format|
-      format.js do
-        if @plan.persisted?
-          render :create
-        else
-          render :new
+      if @plan.type == 'ApplicationPlan' # Only Application plans are implemented in React right now
```

Comment:
Does this work instead?

```suggestion
      case @plan
      when ApplicationPlan # Only Application plans are implemented in React right now
```

Reply:
On the other hand, if the idea of this conditional is to separate old and new implementation and eventually the old one will be removed, then please use a safe-guard clause instead:

```ruby
  unless @plan.type == 'ApplicationPlan'
    ThreeScale::Deprecation.warn "Plans are being migrated to React and this will no longer be used"

    return format.js do
      render @plan.persisted? ? :create : :new
    end
  end

  json = @plan.persisted? ? { notice: 'Plan copied.' } : { error: 'Plan could not be copied' }
  json[:plan] = @plan.decorate.index_table_data.to_json
  format.json { render json: json, status: :created }
```

Reply:
It does with a string `when 'ApplicationPlan'`